src/stores/Binding/Store.ts

```diff
@@ -433,47 +509,51 @@ const getInitialState = (
 
         get().resetState(materializationRehydrating);
 
-        if (connectorTagId && connectorTagId.length > 0) {
-            const { data, error } = await getSchema_Resource(connectorTagId);
-
-            if (error) {
-                get().setHydrationErrorsExist(true);
-            } else if (data?.resource_spec_schema) {
-                await get().setResourceSchema(
-                    data.resource_spec_schema as unknown as Schema
-                );
+        const connectorTagResponse = await hydrateConnectorTagDependentState(
+            connectorTagId,
+            get
+        );
 
-                get().setBackfillSupported(!Boolean(data.disable_backfill));
-            }
-        }
+        const fallbackInterval =
+            entityType === 'capture' &&
+            typeof connectorTagResponse?.default_capture_interval === 'string'
+                ? ''
+                : null;
 
         if (editWorkflow && liveSpecIds.length > 0) {
             const { data: liveSpecs, error: liveSpecError } =
                 await getLiveSpecsByLiveSpecId(liveSpecIds[0], entityType);
 
-            if (liveSpecError) {
+            if (liveSpecError || !liveSpecs || liveSpecs.length === 0) {
                 get().setHydrationErrorsExist(true);
-            } else if (liveSpecs && liveSpecs.length > 0) {
-                if (draftId) {
-                    const { data: draftSpecs, error: draftSpecError } =
-                        await getDraftSpecsByDraftId(draftId, entityType);
-
-                    if (draftSpecError) {
-                        get().setHydrationErrorsExist(true);
-                    } else if (draftSpecs && draftSpecs.length > 0) {
-                        get().prefillBindingDependentState(
-                            entityType,
-                            liveSpecs[0].spec.bindings,
-                            draftSpecs[0].spec.bindings
-                        );
-                    }
-                } else {
-                    get().prefillBindingDependentState(
-                        entityType,
-                        liveSpecs[0].spec.bindings
-                    );
-                }
+
+                return Promise.reject(
+                    liveSpecError?.message ??
+                        `An issue was encountered fetching the live specification for this ${entityType}`
+                );
             }
+
+            const draftSpecError = await hydrateSpecificationDependentState(
+                connectorTagResponse?.default_capture_interval,
+                entityType,
+                fallbackInterval,
+                get,
+                liveSpecs[0].spec,
+                searchParams
+            );
+
+            if (draftSpecError) {
+                get().setHydrationErrorsExist(true);
+
+                return Promise.reject(draftSpecError.message);
```

Comment:
Since we do nothing with this promise rejection we should not pass back a value with it.

Reply:
As previously discussed, this message is relayed in the LogRocket console in the hydrator in an effort to aid debugging. The details form store takes a similar approach which can be quite helpful.